### PR TITLE
Added support for Mongoose

### PR DESCRIPTION
**Pull request by papigers**
I've saw [`relay-mongodb-connection`](https://github.com/mikberg/relay-mongodb-connection), loved it and wanted to do the same for mongoose users. Most of the code is basically the same, just few changes to match mongoose API.

Added the function `connectionFromMongooseQuery` and `connectionFromMongooseAggregate` to create a connection from mongoose `Query` or `Aggregate` respectively.

Didn't want to open an individual package, since most of the code is the same and NOT MINE.

Haven't done much testing, just enough to cover all my use cases.

### CODE REVIEW

**Comment by mikberg** [comment]
Very cool!

Would love to have more code reuse, though. Maybe we could abstract some stuff and create adapters for the respective situations. Also, not feeling confident in merging without any test coverage on the new functions.

**Comment by papigers** [comment]
I understand. I'll see if I can make some time to make a more decent PR, too busy at the moment unfortunately. 

I'll also try to refactor some repetitive code.

File: package.json [hunk 1]
```diff
@@ -3,34 +3,39 @@
...
-  "repository" : {
-    "type" : "git",
-    "url" : "https://github.com/mikberg/relay-mongodb-connection.git"
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/mikberg/relay-mongodb-connection.git"
   },
   "scripts": {
-    "compile": "mkdir -p lib && babel index.js --out-file lib/index.js",
+    "compile": "webpack --colors --progress -p",
```
Comment (mikberg) [inline package.json]: why use webpack? If anything, I think we should aim to remove a build step alltogether

File: package.json [hunk 2]
```diff
@@ -3,34 +3,39 @@
...
+    "mongoose"
   ],
   "author": "Mikael Berg",
   "license": "MIT",
   "devDependencies": {
     "babel": "^5.8.34",
-    "babel-core": "^5.8.34",
+    "babel-core": "^5.8.38",
     "babel-eslint": "^4.1.5",
+    "babel-loader": "^6.2.4",
```
Comment (mikberg) [inline package.json]: Babel tools should be at the same version

File: src/connectionFromMongoCursor.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+export default async function connectionFromMongoCursor(inMongoCursor, args = {}, mapper) {
+  const mongodbCursor = inMongoCursor.clone();
+  const count = await mongodbCursor.count();
+
+  const offsets = getOffsetsFromArgs(args, count);
+
+  const {
+    skip,
+    limit
+  } = offsets;
```
Comment (mikberg) [inline src/connectionFromMongoCursor.js]: `const { skip, limit } = getOffsetsFromArgs(args, count);`

File: test/db.js [hunk 4]
```diff
@@ -1,5 +1,14 @@
 import { MongoClient } from 'mongodb';
+import mongoose from 'mongoose';
+
+const MONGO_URL = process.env.MONGO_CONNECTION_STRING || 'mongodb://localhost/rmc-test';
 
 export const connect = async () => {
-  return MongoClient.connect(process.env.MONGO_CONNECTION_STRING);
```
Comment (mikberg) [inline test/db.js:4]: Ok, probably better to use `MONGO_URL`, but then we should use it consistently.

**Comment by mikberg** [comment]
Starting to look pretty good! :) Great work!

**Comment by papigers** [comment]
Should be good now :+1:

File: package.json (hunk 1 above)
Comment (papigers) [inline package.json]: Didn't remove the build step, but webpack is gone now.

**Comment by papigers** [comment]
Should probably change version in `package.json` and update changelog.

**Comment by mikberg** [comment]
Updated both now. Release is on it's way. Thanks! 👍

**Comment by papigers** [comment]
The new release doesn't work, babel doesn't compile it right - the compiled version still includes es6 `import` and `export`. It seems that babel didn't compile with the es2015 preset.

**Pull request merged**